### DISCUSSION TITLE
[False information] Documentation recommends to use async with void

### DISCUSSION THREAD

**Discussion by TeaTime762** [PrismLibrary/Prism#2783] [General]
### Description

As far as I know using async with void is bad because it does not catch exception then.
The documentation says in the chapter Commanding in the subsection "Implementing a Task-Based DelegateCommand" to use that.
What am I missing?

### Steps to Reproduce

See documentation

### Platform with bug

Prism Core

### Affected platforms

I was *not* able test on other platforms

**Comment by brianlagunas** [comment]
Commands are considered events and should be treated as such. This is why it's important to add proper error handling and don't let unhandled exceptions occur in your app.

**Reply by TeaTime762** [reply]
Thanks for your answer. Now I understand. I did not know that before. I found further explanation on stackoverflow:
https://stackoverflow.com/questions/12144077/async-await-when-to-return-a-task-vs-void

**Reply by Haukinger** [reply]
It would be very nice from a testing perspective if `DelegateCommand` provided a `Task ExecuteAsync()` method and accepted a `Func<Task>` as alternative to `Action` so one can write

    MyCommand = new( DoStuffAsync );

instead of

    MyCommand = new( async () => await DoStuffAsync() );

and in the test

    await viewModelUnderTest.MyCommand.ExecuteAsync();

instead of

     await viewModelUnderTest.DoStuffAsync(); // needs to be at least internal instead of private

so that we reduce the surface of the view model as well as covering more in the test, i.e. that `MyCommand` is actually setup to call `DoStuffAsync`.

**Reply by brianlagunas** [reply]
This has been talked about at length in the past. The ICommand interface is not Async. Prism will not be adding this type of command. You can use another command from another project if you would like this behavior. This video covers another strategy you could use to handle async void https://youtu.be/O1Tx-k4Vao0
